codec-http2/src/test/java/io/netty/handler/codec/http2/Http2FrameCodecTest.java

```diff
@@ -150,22 +149,26 @@ public void entityRequestEntityResponse() throws Exception {
         // Release hello to emulate ByteToMessageDecoder
         hello.release();
         Http2DataFrame inboundData = inboundHandler.readInbound();
-        assertEquals(releaseLater(new DefaultHttp2DataFrame(bb("hello"), true, 31).streamId(stream.id())),
-                     releaseLater(inboundData));
+        Http2DataFrame expected = new DefaultHttp2DataFrame(bb("hello"), true, 31).streamId(stream.id());
+        assertEquals(expected, inboundData);
         assertEquals(1, inboundData.refCnt());
+        expected.release();
         assertNull(inboundHandler.readInbound());
 
         inboundHandler.writeOutbound(new DefaultHttp2HeadersFrame(response, false).streamId(stream.id()));
         verify(frameWriter).writeHeaders(eq(http2HandlerCtx), eq(1), eq(response), anyInt(),
```

Comment:
@normanmaurer - In other tests we don't use `spy` and instead use annotations to indicate mock objects [1] and re-initilize these mock objects for each test run. In my experience this approach does not throw random exceptions while debugging the code that I have seen when we use wrapping an existing object with `spy()`. So "ditching this approach" means to not use `spy` but instead use the approach defined by [1]+[2] or just don't use mocks.

[1] https://github.com/netty/netty/blob/4.1/codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java#L75
[2] https://github.com/netty/netty/blob/4.1/codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java#L119